src/components/ToyList.js

```diff
@@ -1,34 +1,43 @@
-import React, { useState } from "react";
-import { Link } from "react-router-dom";
-import { Container, Card, Row, Col, Button } from "react-bootstrap";
-import { sortToys, filterReview } from "../helpers/toyHelpers";
+import React, { useState } from 'react';
+import { Link } from 'react-router-dom';
+import { Container, Card, Row, Col, Button } from 'react-bootstrap';
+import { sortToys, filterReview, searchToys } from '../helpers/toyHelpers';
+import Search from './SearchToys';
 
 const ToyList = ({ toys }) => {
-  const [sortDirection, setSortDirection] = useState("forward");
-  const [reviewFilter, setFilter] = useState("off");
+  const [sortDirection, setSortDirection] = useState('forward');
+  const [reviewFilter, setFilter] = useState('off');
+  const [searchInput, setSearchInput] = useState('');
 
   function HandleSortOptionChange() {
-    setSortDirection("forward");
+    setSortDirection('forward');
   }
 
   function HandleSortOptionChangeBackward() {
-    setSortDirection("backward");
+    setSortDirection('backward');
   }
 
   function HandleFilterWithReviews() {
-    setFilter("on");
+    setFilter('on');
+  }
+
+  function getSearchInput(input) {
+    setSearchInput(input);
   }
 
   function ResetFilter() {
-    setFilter("off");
+    setFilter('off');
   }
 
   let toyArray = toys ? sortToys(toys, sortDirection) : null;
   let filteredToys = filterReview(toyArray, reviewFilter);
+  let searchedToys = searchToys(toys, searchInput);
+  filteredToys = searchedToys;
```

Comment:
What about making `searchToys(...)` a selector, and pass the following arguments to it:
- sortDirection
- reviewFilter
- searchInput

Then can have a single selector that handles all that (and don't need to pass toys around a bunch of places).

Reply:
Something like this:

Usage:
```
 const toys = useSelector(searchToys(sortDirection, reviewFilter, searchInput));
```

Code:
```
export const searchToys = (sortDirection, reviewFilter, searchTerm) => (
  ({ toys }) => {
    let results = Object.assign([], toys);

    if (searchTerm)
      results = results.filter(toy => toy.name.toLowerCase().includes(searchTerm.toLowerCase()))

    if (reviewFilter === 'on')
      results = results.filter(toy => toy.reviews.length > 0);

    results.sort((toyA, toyB) => toyA.name.localeCompare(toyB.name));
 
    return sortDirection === 'backward' ? results.reverse() : results;
  }
)
```

Reply:
If `sortDirection`, `reviewFilter`, and `searchTerm` were stored in redux state, then you wouldn't need to pass them through to the selector (the selector can fetch them directly from redux).

Reply:
Hey Shawn, 

I did fixed everything before this recommendation. I am confused as to how I am supposed to go about doing this part? Can you provide more clarity on the recommendation and how do you go about managing the state for the `sortDirection`, `reviewFilter`, and` searchInput` if I were to use the `useSelector`?

Reply:
You don't have to use a selector, was just offering up some suggestions.  Anyway, if you want to update the redux state, you'd dispatch actions, then you'd have reducers listening for the actions, and update the state, similar with what you did your toys and reviews.

For example, let's say you want to store the reviewFilter value inside of redux state rather than internal component state.   Instead of using seState to generate a setFilter() function to update the internal component state, you'd create an action and would dispatch that action instead:

```
const reviewFilterAction = (value = 'off') => { type: 'SET_REVIEW_FILTER', payload: value };
```

Then when you click button to turn on the filter, on the onClick would be `dispatch(reviewFilterAction('on'))`.
And when you click the reset button, onClick would be `dispatch(reviewFilterAction('off'))`.

Then you'd set up a reducer to handle the `SET_REVIEW_FILTER`.  You'd probably want to set up a separate reducer file, but to keep things a bit simple, let's just add to toyReducer.js, let's add a new key named `displayOptions` (for lack of a better name), and set up an initial state for it in the reducer:

```
export const initialState = {
  toys: [],
  displayOptions: {
    reviewFilter: 'off',
    sortDirection: 'forward',
    searchTerm: ''
  }
};
```

Then, we just need another case statement to handle the new type of `SET_REVIEW_FILTER`:

```
 case 'SET_REVIEW_FILTER':
   return { ...state, displayOptions: { ...state.displayOptions, reviewFilter: action.payload };
```

You'd then just do the same thing for the sortDirection and the searchTerm.

Reply:
Then to fetch the toys, you'd need to tweak the selector a bit so the values come from the state, rather than being sent in:

```
export const searchToys = ({ toys, displayOptions: { reviewFilter, sortDirection, searchTerm }) => {
  let results = Object.assign([], toys);

  if (searchTerm)
    results = results.filter(toy => toy.name.toLowerCase().includes(searchTerm.toLowerCase()))

  if (reviewFilter === 'on')
    results = results.filter(toy => toy.reviews.length > 0);

  results.sort((toyA, toyB) => toyA.name.localeCompare(toyB.name));

  return sortDirection === 'backward' ? results.reverse() : results;
};

```

And then in ToyList, you'd use this to fetch all the filtered/sorted/searched toys, by using the updated selector:

```
const toys = useSelector(searchToys);
```

Reply:
Hey Shawn! I went ahead and didn't mess with the reducers, instead I focused on the first solution you mentioned. Maybe I am missing something, but it looks like I should be able to delete the helpers now that I have the necessary code in the selectors, right?

Reply:
Yeah, if you're using the selector, then those helpers are no longer needed.  Also, notice that you have a merge conflict with the `toySelectors.js` file in this branch.  In other words, one of your other PRs must have had changes and was merged into `main`, then you have changes to toySelectors.js in this branch.  

To fix, when you're in a clean state in `created-searchbar` branch, you should be able to pull in `origin/main` in this branch, address the merge conflicts, commit the changes, then push those commits back to `created-searchbar` remote branch.

Reply:
Thank you! Yay! 

I just now followed those directions. Does it all look right to you?

Reply:
Merge looks good! \o/

Can you delete the `test/helpers/toyHelpers.test.js` file as well, since they are no longer needed:
```
 FAIL  src/test/unit/helpers/toyHelpers.test.js
  ● Test suite failed to run

    Cannot find module '../../../helpers/toyHelpers' from 'unit/helpers/toyHelpers.test.js'

    > 1 | import { sortToys, filterReview } from '../../../helpers/toyHelpers';
        | ^
      2 | import { toys, toyWithReview, toyWithoutReview } from '../../fixtures/toys';
      3 |
      4 | describe('toyHelpers', () => {

      at Resolver.resolveModule (../../node_modules/@jest/core/node_modules/jest-resolve/build/resolver.js:324:11)
      at Object.<anonymous> (unit/helpers/toyHelpers.test.js:1:1)

```

Reply:
Woot, thank you! 

Oh yeah for sure, that makes sense haha. I just deleted it!